pkg/library/libraryapplyconfiguration/types.go

```diff
@@ -0,0 +1,54 @@
+package libraryapplyconfiguration
+
+import (
+	"slices"
+	"strings"
+)
+
+type ApplyConfigurationRunResult struct {
+	ControllerResults []ControllerRunResult `json:"controllerResults"`
+}
+
+type ControllerRunResult struct {
+	ControllerName string              `json:"controllerName"`
+	Status         ControllerRunStatus `json:"status"`
+	Errors         []ErrorDetails      `json:"errors,omitempty"`
```

Comment:
I was going to suggest including the errors themselves here, but this makes sense for reporting purposes